packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts

```diff
@@ -227,9 +227,12 @@ export function createOdspNetworkError(
 			break;
 		case 401:
 		case 403:
-			// The server throws 403 status code with innerMostError code as "serviceReadOnly" for cases where the
-			// database on server becomes readonly. The driver retries for such cases with exponential backup logic.
-			if (innerMostErrorCode === OdspServiceReadOnlyErrorCode) {
+			// After a redirect due to a domain change, auth headers are not sent by the browser and can trigger a 403.
+			if (response?.redirected) {
```

Comment:
The 403 will be the response on the request against the new domain, and yes `response.url` contains the new URL rather than the old one on this scenario.

On these calls, neither the 308 nor 403 responses contain those redirectLocation error codes. 403 is a `System.UnauthorizedAccessException: Access is denied` error